analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,859 @@
+
+.. data:: extensions
+
+    **Required** String containing extensions separated by ``|`` for example ``"mhj|docx"``
+
+
+Example of usage::
+
+    _paq.push(["addDownloadExtensions", "mhj|docx"]);
+
+This list can be rewrote using ``setDownloadExtensions`` function::
+
+    _paq.push(["setDownloadExtensions", extensions]);
+
+.. data:: extensions
+
+    **Required** String containing extensions separated by ``|`` for example ``"7z|apk|mp4"``
+
+
+Example of usage::
+
+    _paq.push(["setDownloadExtensions", "7z|apk|mp4"]);
+
+To track clicking a link as an download using css class simply add ``piwik_download`` class to link element.
+
+This class name can be changed, use ``setDownloadClasses`` to define which CSS class should be tracked::
+
+    _paq.push(["setDownloadClasses", className]);
+
+
+.. data:: className
+
+    **Required** String with css class that should be tracked instead of ``piwik_download``
+
+Example of usage::
+
+    _paq(["setDownloadClasses", "track-this-link-for-download"]);
+
+
+If you want to use JS instead you can add ``trackLink`` function to element ``onClick`` attribute::
+
+    _paq.push(["trackLink", linkAddress, "download", dimensions]);
+
+.. data:: linkAddress
+
+    **Required** Address that link points to.
+
+.. data:: dimensions
+
+    **Optional**  :ref:`Custom dimension<Custom Dimensions>` that should be tracked with this action. Can be multiple dimensions.
+    Written as object property using ``dimension{ID}`` notation.
+
+    Example::
+
+        {
+           dimension1: "example value",
+           dimension2: "example value"
+        }
+
+Example of usage
+
+.. code-block:: html
+
+    <button onClick = function(){_paq.push(["trackLink", "http://www.example.com/example.xrt", "download"]);}>
+        Click me!
+    </button>
+
+
+After each outbound link there is small time frame after which the file will download that will
+ensure there is enough time to track download.
+That time frame is set to 500ms by default. To modify it you can use ``setLinkTrackingTimer`` function::
+
+    _paq.push(["setLinkTrackingTimer" time]);
+
+.. data:: time
+
+    **Required** Time in ms between user interaction and downloading file.
+
+
+Disabling tracking
+``````````````````
+
+You can disable download and outlink tracking for links using CSS classes, simply add ``piwik_ignore`` css class.
+
+To disable using CSS class you can use ``setIgnoreClassess`` function::
+
+    _paq.push(["setIgnoreClasses", className);
+
+.. data:: className
+
+    **Required** Css class name that will be ignored
+
+
+User ID Management
+^^^^^^^^^^^^^^^^^^
+User ID enables merging user data that is collected between many devices and browsers.
+
+You must provide unique user-id for every user. To set user ID for tracked data use ``setUserId`` function::
+
+    _paq.push(["setUserId", userID]);
+
+.. data:: userID
+
+    **Required** Unique, non empty string preserved for each user.
+
+Tracking domains and subdomains
```

Comment:
This section looks like good candidate to move to separate document describing in full manual snippet installation since those configs will be a lot less popular.

Reply:
I think this should still be in the same document for now.